ibm/service/power/resource_ibm_pi_volume_bulk.go

```diff
@@ -0,0 +1,537 @@
+}
+
+func resourceIBMPIVolumeBulkCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sess, err := meta.(conns.ClientSession).IBMPISession()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	name := d.Get(Arg_VolumeName).(string)
+	size := int64(d.Get(Arg_VolumeSize).(float64))
+	var shared bool
+	if v, ok := d.GetOk(Arg_VolumeShareable); ok {
+		shared = v.(bool)
+	}
+	cloudInstanceID := d.Get(Arg_CloudInstanceID).(string)
+	body := &models.MultiVolumesCreate{
+		Name:      &name,
+		Shareable: &shared,
+		Size:      &size,
+	}
+	body.Count = int64(d.Get(Arg_Count).(int))
+	if v, ok := d.GetOk(Arg_VolumeType); ok {
+		volType := v.(string)
+		body.DiskType = volType
+	}
+	if v, ok := d.GetOk(Arg_VolumePool); ok {
+		volumePool := v.(string)
+		body.VolumePool = volumePool
+	}
+	if v, ok := d.GetOk(Arg_ReplicationEnabled); ok {
+		replicationEnabled := v.(bool)
+		body.ReplicationEnabled = &replicationEnabled
+	}
+	if v, ok := d.GetOk(Arg_ReplicationSites); ok {
+		if d.Get(Arg_ReplicationEnabled).(bool) {
+			body.ReplicationSites = flex.FlattenSet(v.(*schema.Set))
+		} else {
+			return diag.Errorf("Replication (%s) must be enabled if replication sites are specified.", Arg_ReplicationEnabled)
+		}
+	}
+	if ap, ok := d.GetOk(Arg_AffinityPolicy); ok {
+		policy := ap.(string)
+		body.AffinityPolicy = &policy
+
+		if policy == "affinity" {
+			if av, ok := d.GetOk(Arg_AffinityVolume); ok {
+				afvol := av.(string)
+				body.AffinityVolume = &afvol
+			}
+			if ai, ok := d.GetOk(Arg_AffinityInstance); ok {
+				afins := ai.(string)
+				body.AffinityPVMInstance = &afins
+			}
+		} else {
+			if avs, ok := d.GetOk(Arg_AntiAffinityVolumes); ok {
+				afvols := flex.ExpandStringList(avs.([]interface{}))
+				body.AntiAffinityVolumes = afvols
+			}
+			if ais, ok := d.GetOk(Arg_AntiAffinityInstances); ok {
+				afinss := flex.ExpandStringList(ais.([]interface{}))
+				body.AntiAffinityPVMInstances = afinss
+			}
+		}
+
+	}
+	if v, ok := d.GetOk(Arg_UserTags); ok {
+		body.UserTags = flex.FlattenSet(v.(*schema.Set))
+	}
+
+	client := instance.NewIBMPIVolumeClient(ctx, sess, cloudInstanceID)
+	vols, err := client.CreateVolumeV2(body)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	// id is a combination of the cloud instance id and all of the volume ids
+	id := cloudInstanceID
+	for _, vol := range vols.Volumes {
+		id += "/" + *vol.VolumeID
+	}
+	d.SetId(id)
+
+	for _, vol := range vols.Volumes {
+		volumeid := *vol.VolumeID
+		_, err = isWaitForIBMPIVolumeBulkAvailable(ctx, client, volumeid, d.Timeout(schema.TimeoutCreate))
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		if _, ok := d.GetOk(Arg_UserTags); ok {
+			oldList, newList := d.GetChange(Arg_UserTags)
+			err := flex.UpdateGlobalTagsUsingCRN(oldList, newList, meta, string(vol.Crn), "", UserTagType)
+			if err != nil {
+				log.Printf("Error on update of volume (%s) pi_user_tags during creation: %s", volumeid, err)
+			}
+		}
```

Comment:
Could you add a check for empty or nil crn ?Which ever applies

Reply:
added nil check